modules/theme-tools/content-options/featured-images-fallback.php

```diff
@@ -0,0 +1,157 @@
+ * @param int          $post_id           The post ID to check.
+ * @param int          $post_thumbnail_id The ID of the featured image.
+ * @param string       $size              The image size to return, defaults to 'post-thumbnail'.
+ * @param string|array $attr              Optional. Query string or array of attributes.
+ *
+ * @return string      $html              Thumbnail image with markup.
+ */
+function jetpack_featured_images_fallback_get_image( $html, $post_id, $post_thumbnail_id, $size, $attr ) {
+	$opts = jetpack_featured_images_get_settings();
+
+	if ( ! empty( $html ) || (bool) 1 !== (bool) $opts['fallback-option'] ) {
+		return trim( $html );
+	}
+
+	if ( jetpack_featured_images_should_load() ) {
+		if ( ( true === $opts['archive'] && ( is_home() || is_archive() || is_search() ) && ! $opts['archive-option'] )
+			|| ( true === $opts['post'] && is_single() && ! $opts['post-option'] ) || ! $opts['fallback-option'] ) {
+				return trim( $html );
+		}
+	}
+
+	if ( class_exists( 'Jetpack_PostImages' ) ) {
+		global $_wp_additional_image_sizes;
+
+		$args = array(
+			'from_thumbnail'  => false,
+			'from_slideshow'  => true,
+			'from_gallery'    => true,
+			'from_attachment' => false,
+		);
+
+		$image = Jetpack_PostImages::get_image( $post_id, $args );
+
+		if ( ! empty( $image ) ) {
+			$image['width']  = '';
+			$image['height'] = '';
+			$image['crop']   = '';
+
+			if ( array_key_exists( $size, $_wp_additional_image_sizes ) ) {
+				$image['width']  = $_wp_additional_image_sizes[ $size ]['width'];
+				$image['height'] = $_wp_additional_image_sizes[ $size ]['height'];
+				$image['crop']   = $_wp_additional_image_sizes[ $size ]['crop'];
+			}
+
+			$image_src = Jetpack_PostImages::fit_image_url( $image['src'], $image['width'], $image['height'] );
+
+			// Use the theme's crop setting rather than forcing to true
+			$image_src = add_query_arg( 'crop', $image['crop'], $image_src );
+
+			$html      = '<img src="' . esc_url( $image_src ) . '" title="' . esc_attr( strip_tags( get_the_title() ) ) . '" class="attachment-' . esc_attr( $size ) . ' wp-post-image" />';
+
+			return trim( $html );
+		}
+	}
+
+	return trim( $html );
+}
+add_filter( 'post_thumbnail_html', 'jetpack_featured_images_fallback_get_image', 10, 5 );
+
+/**
+ * Get URL of one image from a specified post in the following order:
+ * Featured Image then first image from the_content HTML
+ *
+ * @param int          $post_id           The post ID to check.
+ * @param int          $post_thumbnail_id The ID of the featured image.
+ * @param string       $size              The image size to return, defaults to 'post-thumbnail'.
+ *
+ * @return string|null $image_src         The URL of the thumbnail image.
+ */
+function jetpack_featured_images_fallback_get_image_src( $post_id, $post_thumbnail_id, $size ) {
+	$image_src = ( ! empty( wp_get_attachment_image_src( $post_thumbnail_id, $size )[0] ) ) ? wp_get_attachment_image_src( $post_thumbnail_id, $size )[0] : null;
```

Comment:
This syntax is only supported since PHP 5.4 and Travis is failing due to this.
```
PHP Parse error:  syntax error, unexpected '[', expecting ')' in ./modules/theme-tools/content-options/featured-images-fallback.php on line 78
```